### PR TITLE
test: add pipeline and async behavior tests

### PR DESCRIPTION
**Pull request by Mariomsoliman**
What’s included
Tests for Whisper error handling (invalid paths, missing files, edge cases)
Basic async upload behavior tests (upload returns quickly, status updates over time)
Simple scripts to manually verify pipeline execution
What this verifies
Upload endpoint responds without blocking
Pipeline runs and updates session status correctly
Errors are surfaced clearly during transcription
Current pipeline flow works end-to-end (based on existing API)
Note

This PR intentionally avoids testing transcript segments or migration behavior, since those are not yet present on main. Those tests will be added in a follow-up PR once the implementation is merged.

### CODE REVIEW

File: backend/test_pipeline_integration.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+    segments = result["segments"]
+
+    print(f"[test] ✅ Pipeline output valid")
+    print(f"[test]    - Text length: {len(transcript_text)} characters")
+    print(f"[test]    - Segments: {len(segments)}")
+
+    # Save to database with segments
+    print("[test] Step 5: Saving transcript with segments to database...")
+    try:
+        save_transcript(session_id, transcript_text, segments)
```
Comment (undralg) [inline backend/test_pipeline_integration.py]: This test is exercising an API that current main does not provide: `save_transcript` takes `(session_id, transcript_text)` and the `transcripts` table has no `segments` column. `process_recording` also currently returns transcript text, not a `{text, segments}` dict. Please update the implementation/schema first, or adjust the tests to the current transcript API, otherwise these tests will fail as soon as backend tests run in CI.

**Review by undralg — CHANGES_REQUESTED** [review]
Thanks for adding coverage here. Requesting changes because the new tests currently target transcript segments behavior that is not present on main yet, so they will fail once backend tests are run against the current service/schema.

**Comment by Mariomsoliman** [comment]
Thanks, that makes sense. I updated this PR to remove tests that depended on transcript segments/migration since that’s not on main yet.

Now the PR only includes tests that match current behavior (Whisper errors + basic async flow).

I’ll add the segments/migration tests in a follow-up PR once that implementation is merged.
